src/main/java/org/opensearch/security/dlic/rest/api/RolesApiAction.java

```diff
@@ -77,4 +79,22 @@ protected CType getConfigName() {
         return CType.ROLES;
 	}
 
+	@Override
+	protected boolean hasPermissionsToCreate(final SecurityDynamicConfiguration<?> dynamicConfiguration, final Object content, final String resourceName) throws IOException {
+		if (restApiAdminPrivilegesEvaluator.containsRestApiAdminPermissions(content)) {
```

Comment:
If the user is assigned `restapi:admin/*` what is the purpose of the `isSuperAdmin()` check? I saw that the implementation of `isSuperAdmin` was also changed in this PR, but it has a specific meaning in the security plugin.

Curious, why not follow a similar pattern to cluster permissions where in `ConfigModelV7` and `ConfigModelV6` there is a WildcardMatcher that tests the given action to see if it matches permissions that the user is assigned.

See https://github.com/opensearch-project/security/blob/main/src/main/java/org/opensearch/security/securityconf/ConfigModelV7.java#L544-L546

Reply:
Well this is due to that fact of:
```yaml
  cluster_permissions:
    - "*"
```
with the star. If splitting up of roles is a good solution then I will use he same method. As I mentioned before I do not like solution I made to check such permissions.